hermes-client/src/main/java/pl/allegro/tech/hermes/client/HermesClientBasicRetryCondition.java

```diff
@@ -7,7 +7,7 @@
 public class HermesClientBasicRetryCondition implements Predicate<HermesResponse> {
     @Override
     public boolean test(HermesResponse response) {
-        return isClientTimeoutOrServerError(response) || isFailedExceptionally(response);
+        return (response != null) && (isClientTimeoutOrServerError(response) || isFailedExceptionally(response));
```

Comment:
if i get it right: if response is null, we will not retry. i think we should retry even if response is null. btw what caused this in your case?

so it would be:

```
response == null || ....
```